CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+# Contributing
+
+## LAPIS Developer Documentation
+
+This section is a high-level overview to help you understand the LAPIS codebase.
+It is intended for developers who want to contribute to LAPIS.
+
+### LAPIS' Architecture
+
+LAPIS is divided into three main components (from outer to inner):
+
+* the HTTP layer
+* the query mapping layer
+* the SILO client layer
+
+#### HTTP Layer
+
+The HTTP layer is responsible for handling incoming requests and sending responses.
+It mainly relies on Spring.
+It is responsible for:
+
+* parsing the incoming request and handling different request types
+    * GET
+    * POST with JSON body
+    * POST with form data
+* handling response data formats, such as JSON and CSV
+* setting response headers, such as for the "Content-Disposition" header for the "download as file" feature
+* response compression
+* catching exceptions and making sure that LAPIS returns appropriate HTTP status codes
+
+LAPIS makes extensive use of Spring hooks such as the `OncePerRequestFilter` to implement some of these features.
+This approach has the advantage that it doesn't need to be implemented for every endpoint individually.
+
+#### Query Mapping Layer
+
+Once the data is extracted from the request, it is passed to the query mapping layer.
+This layer maps the request to a SILO query.
+A SILO query consists of two parts:
+
+* the **query action** (aggregated, details, etc.): This is directly determined by which **LAPIS endpoint** has been
+  called.
+  LAPIS offers endpoints for every SILO query action.
+* the **query filter**: This is determined by the **request parameters**.
+  LAPIS tries to match the request parameters on the metadata fields defined in the database configuration.
+
+There are some concepts that we use throughout the LAPIS code:
+
+* Metadata fields are defined by the instance maintainer in the database configuration.
+* Sequence filters are derived from the metadata fields.
+  They contain the plain metadata fields (such as `country` or `date`)
+  and the derived filters (such as `country.regex` or `dataFrom` and `dateTo`).
```

Comment:
I tried to improve the phrasing here but not sure whether I succeeded, my attempt:

> Available filters depend on the types of metadata fields.
Filters that match metadata field names (such as `country` or `date`) perform equality checks, while others support advanced filtering such as matching regular expressions or range filtering (such as `country.regex`, `dataFrom`, or `dateTo`).